sdks/go/pkg/beam/runners/dataflow/dataflowlib/job.go

```diff
@@ -239,6 +252,13 @@ func Translate(ctx context.Context, p *pipepb.Pipeline, opts *JobOptions, worker
 
 // Submit submits a prepared job to Cloud Dataflow.
 func Submit(ctx context.Context, client *df.Service, project, region string, job *df.Job) (*df.Job, error) {
+	if job.ReplaceJobId != "" {
```

Comment:
A cleaner way to do this would be to add replaceJobId as a boolean parameter here. We already have the JobOptions in the only thing calling this - https://github.com/jrmccluskey/beam/blob/ba8ffc295e49985ef93215808d4cd748845f038b/sdks/go/pkg/beam/runners/dataflow/dataflowlib/execute.go#L102

Reply:
Good call-out. Changed.